File: Problem Set 2/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <math.h>
#include <stdlib.h>

// Functions
int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
// Get Text
  string text = get_string("Text: ");


// Get data from text
  float letters = count_letters(text) * 1.0;
  float words = count_words(text) * 1.0;
  float sentences = count_sentences(text) * 1.0;

// Coleman-Liau Formula
  float index = 0.0588 * (letters / words * 100) - 0.296 * (sentences / words * 100) - 15.8;
  int grade = round(index);

// Print Result
  if (grade >= 16)
  {
    printf("Grade 16+\n");
  }
  else if (grade < 1)
  {
    printf("Before Grade 1\n");
  }
  else
  {
    printf("Grade %i\n", grade);
  }
}



// Function count letters
int count_letters(string text)
{
  int letters = 0;

  for (int i = 0, len = strlen(text); i < len; i++)
  {
    if (isalpha(text[i]))
    {
      letters++;
    }
  }
  return letters;
}



// Function Count Words
int count_words(string text)
{
  int words = 1;

  for (int i = 0, len = strlen(text); i < len; i++)
  {
    if (isspace(text[i]))
    {
      words++;
    }
  }

  return words;
}



// Funtion count Sentences
int count_sentences(string text)
{
    int sentences = 0;

    for (int i = 0, len = strlen(text); i < len; i++)
    {
      if (text[i] == '.' || text[i] == '!' || text[i] == '?')
      {
        sentences++;
       }
    }

    return sentences;
}

File: Problem Set 6/sentimental-readability/readability.py
from cs50 import get_string

# Get_string
text = get_string("Text: ")

# Iniciate values
n_letters = 0 * 1.0
n_words = 0 * 1.0 + 1.0
n_sentences = 0 * 1.0

# Loop trought the text and count the values
for i in range(len(text)):
    if text[i].isalpha():
        n_letters += 1

for i in range(len(text)):
    if text[i].isspace():
        n_words += 1

for i in range(len(text)):
    if text[i] == '.' or text[i] == '!' or text[i] == '?':
        n_sentences += 1

L = n_letters / n_words * 100
S = n_sentences / n_words * 100

# Coleman-Liau formula
N = 0.0588 * L - 0.296 * S - 15.8
grade = round(N)

# Print the result
if grade >= 16:
    print("Grade 16+")
elif grade < 1:
    print("Before Grade 1")
else:
    print(f"Grade {grade}")